prusti-viper/src/encoder/mir/types/encoder.rs

```diff
@@ -98,12 +98,10 @@ impl<'p, 'v, 'r: 'v, 'tcx: 'v> TypeEncoder<'p, 'v, 'tcx> {
                 vir::Type::pointer(self.encoder.encode_type_high(*ty)?)
             }
 
-            ty::TyKind::Ref(_, ty, _) => {
-                // TODO: add real lifetime here?
-                let fake_lft = vir::ty::Lifetime {
-                    name: String::from("lft_fake"),
-                };
-                vir::Type::reference(self.encoder.encode_type_high(*ty)?, fake_lft)
+            ty::TyKind::Ref(region, ty, _) => {
+                let lft_name = format!("{}", region);
```

Comment:
The lifetime name probably needs to be a valid Viper identifier.

Reply:
> The lifetime name probably needs to be a valid Viper identifier.

Valid as in not containing any special characters?

Reply:
Yes.